pkg/xds/envoy/clusters/v3/circuit_breaker_configurer.go

```diff
@@ -2,86 +2,34 @@ package clusters
-		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 100}
-	} else {
-		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 0}
-	}
-}
-
-func (c *OutlierDetectionConfigurer) configureLocalErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
-	if local := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetLocalErrors(); local != nil {
-		outlierDetection.ConsecutiveLocalOriginFailure = local.GetConsecutive()
-		outlierDetection.EnforcingConsecutiveLocalOriginFailure = &wrappers.UInt32Value{Value: 100}
-	} else {
-		outlierDetection.EnforcingConsecutiveLocalOriginFailure = &wrappers.UInt32Value{Value: 0}
+	if !c.CircuitBreaker.HasThresholds() {
+		return nil
 	}
-}
-
-func (c *OutlierDetectionConfigurer) configureStandardDeviationDetector(outlierDetection *envoy_cluster.OutlierDetection) {
-	if stdev := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetStandardDeviation(); stdev != nil {
-		outlierDetection.SuccessRateRequestVolume = stdev.GetRequestVolume()
-		outlierDetection.SuccessRateMinimumHosts = stdev.GetMinimumHosts()
-		if factor := stdev.GetFactor(); factor != nil {
-			outlierDetection.SuccessRateStdevFactor = &wrappers.UInt32Value{Value: uint32(factor.GetValue() * 1000)}
-		}
-		outlierDetection.EnforcingSuccessRate = &wrappers.UInt32Value{Value: 100}
-		outlierDetection.EnforcingLocalOriginSuccessRate = &wrappers.UInt32Value{Value: 100} // takes effect only when SplitExternalLocalOriginErrors is true
-	} else {
-		outlierDetection.EnforcingSuccessRate = &wrappers.UInt32Value{Value: 0}
-	}
-}
-
-func (c *OutlierDetectionConfigurer) configureFailureDetector(outlierDetection *envoy_cluster.OutlierDetection) {
-	if failure := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetFailure(); failure != nil {
-		outlierDetection.FailurePercentageRequestVolume = failure.GetRequestVolume()
-		outlierDetection.FailurePercentageMinimumHosts = failure.GetMinimumHosts()
-		outlierDetection.FailurePercentageThreshold = failure.GetThreshold()
-
-		outlierDetection.EnforcingFailurePercentage = &wrappers.UInt32Value{Value: 100}
-		outlierDetection.EnforcingFailurePercentageLocalOrigin = &wrappers.UInt32Value{Value: 100} // takes effect only when SplitExternalLocalOriginErrors is true
-	} else {
-		outlierDetection.EnforcingFailurePercentage = &wrappers.UInt32Value{Value: 0}
+	cluster.CircuitBreakers = &envoy_cluster.CircuitBreakers{
+		Thresholds: []*envoy_cluster.CircuitBreakers_Thresholds{
+			{
+				Priority:           envoy_config_core_v3.RoutingPriority_DEFAULT,
+				MaxConnections:     c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxConnections(),
+				MaxPendingRequests: c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxPendingRequests(),
+				MaxRetries:         c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxRetries(),
+				MaxRequests:        c.CircuitBreaker.Spec.Conf.GetThresholds().GetMaxRequests(),
```

Comment:
According to: https://github.com/kumahq/kuma/blob/ff2e4f2eab0d5da3f08397692a4c69c2b9a9513d/api/mesh/v1alpha1/circuit_breaker.pb.go#L182-L187
`c.CircuitBreaker.Spec.Conf.GetThresholds()` can be nil

Also, maybe we could DRY it?